### PR TITLE
Split jsonrpc2.go file into multiple files

### PR DESCRIPTION
**Pull request by samherrmann**
This merge request moves some of the contents from the `jsonrpc2.go` file into their own designated file. The new files being introduced (excluding test files) are as follows:

* `conn.go`
* `request.go`
* `response.go`

The motive of this change is to make it easier to navigate the code. Without this change, the `jsonrpc2.go` file is 813 lines of code.

### CODE REVIEW

**Comment by samherrmann** [comment]
I understand that the organizational changes being proposed in this PR are very opinionated. Feel free to reject this PR if you rather keep the `jsonrpc2.go` file as is. 🙂

File: conn.go [hunk 1]
```diff
@@ -0,0 +1,460 @@
...
+			}
+		}
+	}()
+
+	if err := c.stream.WriteObject(m); err != nil {
+		return nil, err
+	}
+	return cc, nil
+}
+
```
Comment (samherrmann) [inline conn.go:310]: I included the types that are below this comment in this file because they are only used by `Conn`.

File: internal_test.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package jsonrpc2
+
+import (
+	"encoding/json"
+	"testing"
+)
+
+func TestAnyMessage(t *testing.T) {
```
Comment (samherrmann) [inline internal_test.go:8]: This is the only test that was left in the `object_test.go` file after I moved `TestRequest_MarshalUnmarshalJSON` into `request_test.go` and `TestResponse_MarshalUnmarshalJSON` into `responde_test.go`. I then realized that this is the only internal test file (all other test files contain external tests). So with that realization and the fact that I did not really understand why this file was called `object_test.go` to begin with, I thought it was more intuitive if this file was named `internal_test.go`.

File: conn.go [hunk 3]
```diff
@@ -0,0 +1,460 @@
...
+	return err
+}
+
+// SendResponse sends resp to the peer. It is lower level than (*Conn).Reply.
+func (c *Conn) SendResponse(ctx context.Context, resp *Response) error {
+	_, err := c.send(ctx, &anyMessage{response: resp}, false)
+	return err
+}
+
+func (c *Conn) close(cause error) error {
```
Comment (samherrmann) [inline conn.go:157]: I changed the order of the methods slightly by making sure that public methods come before private methods.

**Review by keegancsmith — APPROVED** [review]
Yup opionated but you deserve it after your recent deep dives :P. LTGM!

**Pull request merged**